File: ContentProvider/ConsumerApp/java/app/src/main/java/com/dicoding/picodiploma/consumerapp/MappingHelper.java
package com.dicoding.picodiploma.consumerapp;

import android.database.Cursor;

import com.dicoding.picodiploma.consumerapp.entity.NoteItem;

import java.util.ArrayList;

import static android.provider.BaseColumns._ID;
import static com.dicoding.picodiploma.consumerapp.db.DatabaseContract.NoteColumns.TITLE;
import static com.dicoding.picodiploma.consumerapp.db.DatabaseContract.NoteColumns.DESCRIPTION;
import static com.dicoding.picodiploma.consumerapp.db.DatabaseContract.NoteColumns.DATE;

public class MappingHelper {

    public static ArrayList<NoteItem> mapCursorToArrayList(Cursor notesCursor) {
        ArrayList<NoteItem> notesList = new ArrayList<>();

        while (notesCursor.moveToNext()) {
            int id = notesCursor.getInt(notesCursor.getColumnIndexOrThrow(_ID));
            String title = notesCursor.getString(notesCursor.getColumnIndexOrThrow(TITLE));
            String description = notesCursor.getString(notesCursor.getColumnIndexOrThrow(DESCRIPTION));
            String date = notesCursor.getString(notesCursor.getColumnIndexOrThrow(DATE));
            notesList.add(new NoteItem(id, title, description, date));
        }

        return notesList;
    }
}

File: ContentProvider/ConsumerApp/kotlin/app/src/main/java/com/dicoding/picodiploma/consumerapp/MappingHelper.kt
package com.dicoding.picodiploma.consumerapp

import android.database.Cursor

import com.dicoding.picodiploma.consumerapp.entity.NoteItem

import java.util.ArrayList

import android.provider.BaseColumns._ID
import com.dicoding.picodiploma.consumerapp.db.DatabaseContract.NoteColumns.Companion.DATE
import com.dicoding.picodiploma.consumerapp.db.DatabaseContract.NoteColumns.Companion.DESCRIPTION
import com.dicoding.picodiploma.consumerapp.db.DatabaseContract.NoteColumns.Companion.TITLE

object MappingHelper {

    fun mapCursorToArrayList(notesCursor: Cursor): ArrayList<NoteItem> {
        val notesList = ArrayList<NoteItem>()

        while (notesCursor.moveToNext()) {
            val id = notesCursor.getInt(notesCursor.getColumnIndexOrThrow(_ID))
            val title = notesCursor.getString(notesCursor.getColumnIndexOrThrow(TITLE))
            val description = notesCursor.getString(notesCursor.getColumnIndexOrThrow(DESCRIPTION))
            val date = notesCursor.getString(notesCursor.getColumnIndexOrThrow(DATE))
            notesList.add(NoteItem(id, title, description, date))
        }

        return notesList
    }
}
